doc/vhpidirect/examples/arrays.rst

```diff
@@ -0,0 +1,52 @@
+.. program:: ghdl
+.. _COSIM:VHPIDIRECT:Examples:arrays:
+
+Arrays
+######
+
+.. _COSIM:VHPIDIRECT:Examples:arrays:matrices:
+
+:cosimtree:`matrices <vhpidirect/arrays/matrices>`
+**************************************************
+
+Constrained multidimensional arrays of doubles/reals
+====================================================
+
+As explained in :ref:`Restrictions_on_foreign_declarations`, unconstrained arrays are represented by a fat pointer in C,
+and it is not suggested to use fat pointers unless it is unavoidable. Hence, it is desirable for matrices which are to
+be represented as multidimensional arrays to be constrained. However, constraining multidimensional arrays in VHDL and
+C separatedly is error prone.
+
+This example shows a general procedure to share constrained multidimensional arrays of any size. Dimensions of a 2D
+matrix of doubles are defined in C and a helper function is used for VHDL to read those values into the declaration of
+an *array of reals* type. Then, the pointer to the matrix (in C) is retrieved as an access (in VHDL), through another
+helper function. For completeness, IEEE's ``fixed_generic_pkg`` package is used to multiply each value with a constant
+using fixed-point formats.
+
+.. HINT::
+  Different types need to be declared in VHDL, to use this approach for passing multiple matrices of differentes
+  sizes. In order to achieve it with a single type in VHDL, it would need to be unconstrained. See :cosimsharp:`3` for
+  work in progress regarding unconstrained types.
+
+Array and AXI4 Stream Verification Components
+=============================================
+
+`VUnit <http://vunit.github.io/>`_ provides an `integer_array <http://vunit.github.io/data_types/integer_array.html>`_
+package with ``load_csv`` and ``save_csv`` functions. Those are used in `Array and AXI4 Stream Verification Components <http://vunit.github.io/examples.html#id9>`_,
+along with AXI4 Stream Verification Components from the `Verification Component Library <http://vunit.github.io/verification_components/user_guide.html>`_,
+to load fata from CSV files to a UUT. While CSVs as intermediate files are useful for integration with Matlab, Octave,
+NumPy, etc., not having an equivalent `real_array` package posses an additional complexity in applications such as DSP or
+machine learning. This is because values to be handled in fixed-point need to be first converted from doubles to integers.
+
+Subdir :cosimtree:`vunit_axis_vcs <vhpidirect/arrays/matrices/vunit_axis_vcs>` of this example contains a modified
+version of `array_axis_vcs <http://vunit.github.io/examples.html#id9>`_, to replace ``integer_array`` (and CSV
```

Comment:
suggestion: contains a modified version of a vunit example ... or... contains a modified version of .. (borrowed from vunit).

not important